src/dom.js

```diff
@@ -12,31 +12,27 @@ import { signalFn } from "./signal";
  * @module dom
  */
 
-/** Holds the root dom nodes with mounted views. */
-let roots = new Map();
-
-/** Holds the disconnectors per signal. */
-let disconnectors = new WeakMap();
-
 /**
- * Mounts the `viewFn` as a child of `root`. Note that there can only be one
- * mounted view per root.
+ * Mounts the result of `viewFn` as a replacement of `root`.
  *
  * The view function is wrapped inside a signal, therefore the view re-computes
  * whenever a state change in any referenced input signal gets detected.
  *
  * As `mount` only tries to make minimal assumptions on how the DOM gets patched
- * a `patchFn` must be provided. What a view function returns is up to the user
- * but must be understand by the patch function. Whenever the result of `viewFn`
- * changes, `patchFn` gets called with the `root` as the first argument, the
- * last result from `patchFn` as the second argument and the result from
- * `viewFn` as the third argument. The patch function must ensure that the DOM
- * gets updated accordingly.
+ * a `patchFn` and `cleanupFn` must be provided. What a view function returns
+ * is up to the user but must be understood by the patch function. Whenever the
+ * result of `viewFn` changes, `patchFn` gets called with the last result from
+ * `patchFn` as the first argument (`root` on first call) and the result from
+ * `viewFn` as the second argument. The patch function must ensure that the DOM
+ * gets updated accordingly. On unmount, the `cleanupFn` gets called with the
+ * last result from `patchFn` and must ensure the previously mounted DOM node
+ * gets removed from the DOM.
  *
  * @param {DOMNode} root The root DOM node.
  * @param {function} viewFn The view function.
  * @param {function} patchFn The patch function.
- * @returns {Signal} The view signal.
+ * @param {function} cleanupFn The cleanup function.
```

Comment:
Do you view the `cleanupFn` as _optional_ - hypothetically, if we mount a component at page load, in a init-function and the lifecycle is bound the page, the cleanup wouldn't be necessary if it's anyway unloaded when the user leaves the page?

Reply:
In my opinion, a `cleanupFn` is never optional, because the cleanup behaviour depends on the mechanism of `patchFn`, about which flyps knows nothing. It is of course possible that the cleanup (unmounting) is a no-op, but this decision must be made by the user.

Reply:
What do you think?

Reply:
For some strange reason, I'm prone to say that I would prefer the no-op implicitly, that is to say the call should not fail or force the `cleanupFn`.

But I cannot say that it would be **correct**.

Reply:
Okay, I agree, `cleanupFn` defaults to no-op now.

**correct**? What is **correct**?